backends/credhub/src/docs/asciidoc/snippets/introduction.adoc

```diff
@@ -36,3 +36,5 @@ Credential responses include a unique identifier in the key 'id'. This ID is a u
 As of 2.0.0, set requests always overwrite the credential that already exists.
 
 As of 2.0.1, generate requests can be set to overwrite, no-overwrite, or converge for the mode parameter. The default mode for generate is converge as of 2.0.0. Converge will only overwrite if the generate request parameters do not match the existing credential.
+
+As of 2.10.0, when the generate requests are set to converge for the mode parameter, converge will not overwrite certificates if duration is the only parameter that does not match the existing certificate credentials.
```

Comment:
We took a guess on the version the new feature will be in. This might need to be updated.

Reply:
Two things:

1. This looks like a breaking change to me. If we ship it as is, I would say we need to bump major.
1. We are trying to stay on the "one version" idea, so that we don't have multiple versions to maintain and patch. I think that we would like this feature to be configurable through a feature flag so that we don't need to bump major and be forced to maintain multiple versions.

Reply:
Yeah this change is what I was [thinking about](https://github.com/pivotal/credhub-release/issues/60#issue-942313486) as well. However, Jesse's [initial proposal](https://github.com/pivotal/credhub-release/issues/60#issue-942313486) argues that accounting for duration in the `converge` mode (aka the existing behavior) is wrong to begin with. I personally don't have context on how consumers of credhub use the `converge` mode, but I believe `bosh` is the only credhub consumer that depends on this feature. So I'm kind of relying on the bosh team to judge whether the old behavior is a bug.

Reply:
As a member of the BOSH team, I think I would consider the old behavior a bug. The behavior of `converge` is meant to cause a credential to regenerate when the properties declared for the variable have changed. For things like a change to common name or signing CA, it makes a lot of sense that converge mode would cause the certificate to be regenerated - the underlying identity of the certificate has changed. This does not seem the case with duration / expiration date.

The cases this would be a breaking change would be as follows:
1. The BOSH operator has a manifest that defines a certificate using `update_mode: converge`.
2. They want to update *only* the duration field in order to either lengthen or shorten its expiration date.
3. They expect that doing a BOSH deploy will automatically regenerate the certificate with `update_mode: converge` for them using the new duration. The changes would cause it to not regenerate in this case.

I am not aware of any product or customer relying on this behavior. It seems very narrow to me, and can be solved by other workflows (such as making an explicit `credhub generate` CLI or equivalent API call). That said, if someone is relying on it, it would be a breaking change for them. 

We added an explicit feature flag on a temporary branch earlier that would default back to the earlier converge behavior unless turned on. It wasn't difficult, but we're holding off on merging it into this PR. Let us know which direction you would like to go in (release a new major, deem this as a bug fix and not worry about it, or add an explicit feature flag) and we'll proceed accordingly.

Reply:
Thanks, @bruce-ricard and I can discuss tomorrow and get back to you.

Meanwhile @ystros, can I ask for your input? It seems like the tension is between:
1. Considering duration is a bug, because when the `converge` was introduced, it was never meant for only extending cert duration, which is a narrow and invalid use case. Hence, the lack of a feature flag does not constitute a breaking change, but a bug fix. CredHub can include the current PR in Credhub 2.10. BOSH can include this in a patch as well. 
2. Using the `converge` mode to only extend cert duration is a valid use case that people relies on. So CredHub should 1) include a feature flag (with the default being the existing behavior) and include this in a CredHub 2.10 or 2) bump major to CredHub 3.0 and BOSH would need to do the same.

What I'm trying to get at, is BOSH team might have better insight on whether using the `converge` mode to only extend cert duration is a valid use case that people relies on (CredHub, AFAIK, does not have a standalone cert use case outside of BOSH). Or to phrase the question a little differently: When the current PR (aka no feature flag) is merged and release, would the resulting new BOSH version be considered having a breaking change & needs major version bump? What would the BOSH team's decision be?

Reply:
OK right, fair enough. I actually talked with Jesse about that and I now remember suggesting that the current behavior is a bug...